### PR TITLE
docs: triage the docs corpus by audience and exclude internal notes (#989)

### LINKED ISSUE #989: DOCS-3: Triage docs by audience and exclude internal material from the build
## Parent

Part of DOCS-0 (#986)

## What to build

Decide who each document under `docs/` is for, and keep internal material out of the published site. Triage everything as published or internal: agent guidance, research notes, and planning/design scratch are internal; tutorials, how-to guides, reference pages, explanation pages, ADRs, validation status, and algorithm references are candidates for publication. Exclude internal material from the Sphinx build so that enabling Markdown cannot publish it accidentally, and record the boundary where contributors will see it. This is the prerequisite for turning on Markdown sources, because a Markdown suffix publishes everything it can find.

## Acceptance criteria

- [ ] Every existing docs page is assigned published or internal, and published pages are assigned their reader-need quadrant
- [ ] Internal material is excluded from the build by path or exclusion pattern, not by convention alone
- [ ] ADRs, vocabulary/context documents, validation status, and algorithm references have explicit published/internal decisions
- [ ] The boundary and its rationale are documented in contributor guidance
- [ ] The build produces zero orphan-document warnings and the DOCS-1 gate is green

## Blocked by

- DOCS-1 (##986)

### PR DESCRIPTION
**Pull request by monocongo**
## Summary

Triage the whole `docs/` corpus by audience and enforce the internal side of the boundary, per DOCS-3 (#989):

- Add `exclude_patterns` entries in `docs/conf.py` for internal working notes: `agent/`, `design/`, `explorer/`, `research/fire-indices-cli-approach.md`, `research/interactive-climate-explorer-landscape.md`, `architecture-deepening-review-*.md`, and `test_fixture_management.md`. Exclusion is by path, not convention, so enabling Markdown sources cannot publish them by accident.
- Add a "Documentation audience" section to `CONTRIBUTING.md`: the published/internal rule, its rationale (the site teaches one version of the truth; working notes drift and are addressed to maintainers), and the current assignment of every page — published pages with their reader-need quadrant, internal pages by path.
- Record explicit decisions for the material the issue calls out: ADRs, `docs/algorithm_refs/`, `docs/wildfire_applications.md`, `docs/xarray_compatibility.md`, the architecture and overview notes, and the root context and validation documents (`src/climate_indices/CONTEXT.md`, `VALIDATION.md`) are published; `CONTEXT-MAP.md`, the Explorer context, the architecture review, and the fixture-management strategy are internal.
- Publish `docs/research/nclimgrid-acquisition-and-redistribution.md`: the troubleshooting guide sends readers there for source provenance and attribution, so it serves the reference audience despite its directory. The other two research notes stay internal.
- State that build infrastructure and unreferenced assets (`docs/gallery/*.png`) are not pages and are outside the split.

Internal pages stay where they are rather than moving out of `docs/`: reference stability for the many repository links into `docs/agent/` and `docs/design/`, and the issue's own acceptance criterion allows exclusion by path. The exclusion list is the single place to update when a page is added to or moves between audiences.

## Deferred to the Markdown-enablement tickets

A scratch dual-suffix build (`myst_parser` plus `source_suffix = [".rst", ".md"]`) surfaced what DOCS-4 (#990) and the navigation tickets have to resolve; all of it is recorded on #990:

- Twenty-plus published Markdown pages have no toctree placement yet, so each emits `toc.not_included` — DOCS-4 with DOCS-7 (#993) and DOCS-8 (#994).
- `docs/adr/0005-fire-module-api.md` links to the internal `docs/design/fire-subsystem.md`, and `docs/development-guide.md` links to the repository-root `CONTRIBUTING`; neither resolves as a Sphinx document.
- `docs/wildfire_applications.rst` and `docs/wildfire_applications.md` collide once both suffixes are live (`multiple files found for the document`), so the wrapper is deleted rather than excluded.
- `docs/xarray_compatibility.md` has two links to a missing anchor.

## Verification

- `uv run --extra docs sphinx-build -E -b html -W --keep-going docs docs/_build/html` — build succeeded, 0 warnings
- `uv run --extra docs sphinx-build -E -b doctest docs docs/_build/doctest` — 66 tests, 0 failures
- Scratch MyST dual-suffix build — no internal page in the discovered set; dropping the new patterns discovers all nine of them, so the exclusion is what keeps them out
- `uv run ruff check src/ tests/` and `uv run ruff format --check src/ tests/` — clean
- `uv run mypy src/ tests/test_type_checking.py` — no issues
- `uv run pytest` — 2127 passed, 423 deselected

Closes #989

## Summary by Sourcery

Establish the documentation audience boundary and prevent internal working notes from appearing on the published site.

Enhancements:
- Define and document the published-versus-internal documentation audience assignments across the repository.
- Exclude internal working notes from Sphinx discovery by path to prevent accidental publication when Markdown support is enabled.

Documentation:
- Publish the documentation audience policy, page classifications, and handling of non-page assets in CONTRIBUTING.md.

### CODE REVIEW

**Comment by monocongo** [comment]
### 🛠️ Multi-Agent Review Summary
- **Review target:** https://github.com/monocongo/climate_indices/pull/1003 (`docs/989-docs-audience-triage`, base `9dc59955`)
- **Lanes run:** 4 read-only subagent lanes — correctness/edge cases, security/content-leak, maintainability, tests/verification
- **Validation:** `sphinx-build -E -b html -W --keep-going docs docs/_build/html` — 0 warnings; `sphinx-build -E -b doctest` — 66 tests, 0 failures; scratch MyST dual-suffix build — internal paths never discovered, and all nine are discovered when the patterns are removed; `ruff check`/`ruff format --check` — clean; `mypy src/ tests/test_type_checking.py` — no issues; `pytest -q` — 2127 passed, 423 deselected
- **Fix commits:** [92fa2a22 — docs: correct the audience triage after review (#989)](https://github.com/monocongo/climate_indices/commit/92fa2a22)

### Findings and fix status
#### [P3] The boundary section named two paths that no longer exist — Fixed
- **Location:** `CONTRIBUTING.md` (build infrastructure and unreferenced-asset lines)
- **Origin:** Correctness, security, maintainability (same claim, three lanes)
- **Problem:** `docs/requirements.txt` (deleted by DOCS-1, the docs extra resolves from `uv.lock`) and `docs/project-scan-report.json` (deleted by DOCS-2) were listed as current infrastructure. The exact stale-fact class this project exists to remove.
- **Applied fix:** both dropped; `docs/make.bat` listed; commit `92fa2a22`.

#### [P3] "Published" was presented as what the site serves today — Fixed
- **Location:** `CONTRIBUTING.md` (published bullet, `docs/pypi_release.rst` entry)
- **Origin:** Maintainability, security, tests
- **Problem:** `source_suffix` is still `.rst`, so 20-plus Markdown pages are repository-readable only, and `docs/pypi_release.rst` was counted as a how-to guide although it is an `:orphan:` URL shim.
- **Applied fix:** the published bullet states that Markdown pages join the site when Markdown sources are enabled, and the orphan is named as a published URL-retention shim with no quadrant; commit `92fa2a22`.

#### [P3] A published page sent readers to an internal note — Fixed
- **Location:** `docs/troubleshooting.rst:901-902` → `docs/research/nclimgrid-acquisition-and-redistribution.md`
- **Origin:** Security/content-leak
- **Problem:** the one published→internal reference in the corpus. The note carries source provenance and attribution constraints for the tutorial data, which is reference material, not a working note.
- **Applied fix:** the note is published (reference) and removed from the exclusion patterns; the other two research notes stay internal; commit `92fa2a22`.

#### [P4] Drift wording covered moves but not additions, and disagreed with its own glob — Fixed
- **Location:** `CONTRIBUTING.md` (boundary paragraph, internal list) vs `docs/conf.py` (`architecture-deepening-review-*.md`)
- **Origin:** Maintainability
- **Problem:** a new internal page was not covered by the instruction, and the prose pinned one dated review file while the pattern already covered later ones.
- **Applied fix:** "added or moves between audiences"; the prose uses the same glob as the config; commit `92fa2a22`.

#### [P2] Markdown-enablement gate hazards left for DOCS-4 — Blocked (scope)
- **Location:** `docs/adr/0005-fire-module-api.md:33`, `docs/development-guide.md:225`, `docs/wildfire_applications.rst:1`, `docs/xarray_compatibility.md:23,79`
- **Origin:** Tests, security, correctness (deduplicated: four hazards, one cause)
- **Problem:** with the dual suffix live and `-W`, these emit `Unknown source document 'design/fire-subsystem'`, an unresolved `CONTRIBUTING` xref, `multiple files found for the document "wildfire_applications"`, and two missing-anchor xrefs; plus `toc.not_included` for every published Markdown page until navigation lands. They do not reproduce on this branch because Markdown sources are not built yet.
- **Applied fix:** **Blocked** — appearing only once Markdown sources are enabled puts them in DOCS-4 (#990) with DOCS-7 (#993)/DOCS-8 (#994), and whether the fire subsystem design becomes published or the link becomes a repository URL is a publication decision for that ticket. Recorded in a tracker comment on #990; PR body updated to list them instead of claiming two.

### Remaining blockers
None for this PR. The audience exclusion is inert until Markdown sources are enabled — that is the documented ordering of this project (triage before MyST), not a defect, and it is now stated in the section itself.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -56,6 +56,49 @@ repository, and concrete in-scope and out-of-scope examples are in
...
+move with the code, are addressed to maintainers, and would otherwise reach
+readers as competing instructions. Because a Markdown suffix publishes every
+page it can find, internal pages are excluded by path in
+`docs/conf.py` (`exclude_patterns`), not by convention: enabling Markdown
+sources cannot leak them. When a page moves between audiences, update that
+exclusion list and this section in the same change.
+
+Current assignments (context documents and validation status live at the
+repository root rather than under `docs/`):
+
```
[the remark this answers is not in the document]
Comment (monocongo) [reply CONTRIBUTING.md]: Addressed in 05b81b79: the parenthetical now names the files — `CONTEXT-MAP.md` and `VALIDATION.md` at the repository root, `src/climate_indices/CONTEXT.md` beside the code. Note the finding had one premise wrong: `VALIDATION.md` is at the repository root, not `src/climate_indices/`.
